### ISSUE TITLE
Low image quality when downscaling

### ISSUE THREAD

**Issue by dleisner** [RazrFalcon/resvg#653]
The use case is a PNG image which is base64-encoded and embedded in the SVG in an `<image>` tag.  In the attached example, the original image is 160x160, and scaled down to 49x49 in the SVG markup.  The rendered image (resvg-rendered-49px.png) quality is degraded, especially on the curves.

When defining the dimensions as 160x160 in the SVG markup, the rendered image (resvg-rendered-160px.png) quality is great.  And when including that image on an HTML page and re-scaling in the HTML markup, `<img src="resvg-rendered-160px.png" width="49" height="49">`, the image quality remains crisp.

We also tried setting the `image-rendering` attribute for `OptimizeQuality`, but the improvement was negligible. 
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/image-rendering

Is is correct that Bicubic interpolation is being used?  Do you have any guidance for improving image quality, and/or are there plans for other interpolation options?

Processing this same image with various interpolation filters in `ImageSharp` (e.g. Lanczos, MitchellNetravali, etc.) does produce a high quality result.

![resvg-rendered-49px](https://github.com/RazrFalcon/resvg/assets/15127801/379e81b0-47cf-4f3c-be88-b67ae7ae469d)
![resvg-rendered-160px](https://github.com/RazrFalcon/resvg/assets/15127801/c7d37682-fa93-4581-afcd-5f4329e20ba5)
![emoji-49px-base64](https://github.com/RazrFalcon/resvg/assets/15127801/98fb643d-ce2e-4844-9a8a-1ff05d00cc1d)

**Comment by RazrFalcon** [comment]
resvg does use bicubic scaling, but I guess something goes wrong. Will see.

**Comment by romanzes** [comment]
Also ran into this issue. The SVG below has an embedded PNG in 1024*1024 size, which looks really aliased when rendered at 100*100 size.
## SVG
![instagram](https://github.com/RazrFalcon/resvg/assets/5735967/395f63ad-61b0-4454-a777-3d36bb199bce)
## resvg output
![instagram](https://github.com/RazrFalcon/resvg/assets/5735967/da7112a3-534f-4688-824c-73d702120600)
## Chromium output
<img width="102" alt="Screenshot 2023-08-24 at 4 35 31 pm" src="https://github.com/RazrFalcon/resvg/assets/5735967/eb26999e-6bb4-4564-bf75-88fa76c39e05">

**Comment by romanzes** [comment]
I also tried explicitly passing `FilterQuality::Bicubic` [here](https://github.com/RazrFalcon/resvg/blob/e128224d9873722b88207dc25a9e20014ac8997d/crates/resvg/src/image.rs#L231), but the result was the same. Could it be a problem with tiny-skia then?

**Comment by RazrFalcon** [comment]
@romanzes It most definitely is.

**Comment by incetarik** [comment]
Any updates on this issue?

**Comment by RazrFalcon** [comment]
It will be done when it will be done.

**Comment by ZhouXiaolin** [comment]
modify the `pipeline` in `tiny-skia` to slightly expand the sparsity of the neighboring 4x4 for `bicubic`, then average them. This might yield some improvement.

**Comment by RazrFalcon** [comment]
I don't think this is the case. It works fine in Skia, meaning the bug is somewhere else.

**Comment by romanzes** [comment]
> I don't think this is the case. It works fine in Skia, meaning the bug is somewhere else.

Actually, Skia has the same problem, embedded rasters look aliased when downscaled. I was able to fix it in my fork of Skia by changing [this line](https://github.com/google/skia/blob/e13594df88a12baf3e1ba7ae8b98feaeb1f5482a/modules/svg/src/SkSVGImage.cpp#L91-L92) to:
```c++
ctx.canvas()->drawImageRect(
            imgInfo.fImage, imgInfo.fDst, SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kLinear));
```
However, as far as I understand, `tiny-skia` doesn't support mipmapping, so this workaround wouldn't help here.

**Comment by RazrFalcon** [comment]
@romanzes That's surprising. I will double check later. If Skia fails in the same way as tiny-skia then there is nothing we can do, sort of. We would have to find a workaround.

And no, tiny-skia doesn't support mipmapping. Afaik it was a lot of work to implement and Skia uses some sort of shared cashing for them. Too much complexity for now.
